llvm/lib/Target/AMDGPU/SIAnnotateControlFlow.cpp

```diff
@@ -305,43 +304,43 @@ bool SIAnnotateControlFlow::handleLoop(BranchInst *Term) {
 }
 
 /// Close the last opened control flow
-bool SIAnnotateControlFlow::closeControlFlow(BasicBlock *BB) {
-  llvm::Loop *L = LI->getLoopFor(BB);
+bool SIAnnotateControlFlow::tryWaveReconverge(BasicBlock *BB) {
```

Comment:
This function is one core part of this change. It would be nice to have more comment with examples before the function header showing when and where wave_converge is inserted.